### PR TITLE
Can access through: Support for SQL rules

### PR DESCRIPTION
**Pull request by AndersGM**
https://app.asana.com/0/1153435032856801/1200004503537768

### CODE REVIEW

File: ruby-gem/lib/api_maker/base_resource.rb [hunk 1]
```diff
@@ -98,7 +98,7 @@ def initialize(ability: nil, api_maker_args: {}, locals:, model:)
     @model = model
   end
 
-  def can_access_through(ability:, relationship:)
+  def can_access_through(ability:, relationship:) # rubocop:disable Metrics/AbcSize
```
Comment (kaspernj) [inline ruby-gem/lib/api_maker/base_resource.rb:101]: Fix the offence instead by doing `can_access_through_has_many` etc. which this method could delegate to?

Comment (AndersGM) [reply ruby-gem/lib/api_maker/base_resource.rb]: 8993ea8

File: ruby-gem/lib/api_maker/base_resource.rb [hunk 2]
```diff
@@ -122,6 +127,57 @@ def can_access_through(ability:, relationship:)
     end
   end
 
+  def handle_empty_conditions(model_class:, reflection:, relationship:, target_model_class:)
```
Comment (kaspernj) [inline ruby-gem/lib/api_maker/base_resource.rb:130]: These methods should be private, right?

**Review by kaspernj — CHANGES_REQUESTED** [review]

File: ruby-gem/lib/api_maker/base_resource.rb (hunk 2 above)
Comment (AndersGM) [reply ruby-gem/lib/api_maker/base_resource.rb:136]: Absolutely, added in 7157430 :-)

File: ruby-gem/spec/api_maker/preloading/has_many_spec.rb [hunk 3]
```diff
@@ -33,7 +33,9 @@
 
     expect(preloader.models).to eq [task]
     expect(sql).to include "JOIN tasks AS accessible_table ON accessible_table.id = tasks.id"
-    expect(sql).to include '(EXISTS (SELECT 1 FROM "tasks" WHERE "tasks"."user_id" = 1 AND (tasks.id = accessible_table.id)))'
+    expect(sql).to include 'EXISTS (SELECT 1 FROM "tasks" WHERE ((' \
```
Comment (kaspernj) [inline ruby-gem/spec/api_maker/preloading/has_many_spec.rb:36]: Consistent double quotes :-) 
```suggestion
    expect(sql).to include "EXISTS (SELECT 1 FROM \"tasks\" WHERE ((" \
```

File: ruby-gem/spec/api_maker/preloading/has_many_spec.rb [hunk 4]
```diff
@@ -33,7 +33,9 @@
...
+      "EXISTS (SELECT 1 FROM account_marked_tasks WHERE account_marked_tasks.task_id = tasks.id AND (account_marked_tasks.id = 5))) OR "\
+      '((tasks.name = \'Some readable task\') OR ("tasks"."user_id" = 1))) AND (tasks.id = accessible_table.id)'
```
Comment (kaspernj) [inline ruby-gem/spec/api_maker/preloading/has_many_spec.rb:38]:
```suggestion
      "((tasks.name = 'Some readable task') OR (\"tasks\".\"user_id\" = 1))) AND (tasks.id = accessible_table.id)"
```

**Review by kaspernj — CHANGES_REQUESTED** [review]

**Pull request merged**